spec.html

```diff
@@ -32,4 +32,106 @@
+        1. If _next_ is *false*, then
+          1. Set _iteratorRecord_.[[Done]] to *true*.
+          1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
+          1. If _remainingElementsCount_.[[Value]] is 0, then
+            1. Let _valuesArray_ be CreateArrayFromList(_values_).
+            1. Perform ? Call(_resultCapability_.[[Resolve]], *undefined*, &laquo; _valuesArray_ &raquo;).
+          1. Return _resultCapability_.[[Promise]].
+        1. Let _nextValue_ be IteratorValue(_next_).
+        1. If _nextValue_ is an abrupt completion, set _iteratorRecord_.[[Done]] to *true*.
+        1. ReturnIfAbrupt(_nextValue_).
+        1. Append *undefined* to _values_.
+        1. Let _nextPromise_ be ? Invoke(_constructor_, `"resolve"`, &laquo; _nextValue_ &raquo;).
+        1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-promise.any-reject-element-functions" title></emu-xref>.
+        1. Let _rejectElement_ be CreateBuiltinFunction(_steps_, &laquo; [[AlreadyCalled]], [[Index]], [[Values]], [[Capability]], [[RemainingElements]] &raquo;).
+        1. Set _rejectElement_.[[AlreadyCalled]] to a new Record { [[Value]]: *false* }.
+        1. Set _rejectElement_.[[Index]] to _index_.
+        1. Set _rejectElement_.[[Values]] to _values_.
+        1. Set _rejectElement_.[[Capability]] to _resultCapability_.
+        1. Set _rejectElement_.[[RemainingElements]] to _remainingElementsCount_.
+        1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] + 1.
+        1. Perform ? Invoke(_nextPromise_, `"then"`, &laquo; _resultCapability_, _rejectElement_[[Reject]] &raquo;).
+        1. Increase _index_ by 1.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-promise.any-reject-element-functions">
+    <h1>`Promise.any` Reject Element Functions</h1>
+    <p>A `Promise.any` reject element function is an anonymous built-in function that is used to reject a specific `Promise.any` element. Each `Promise.any` reject element function has [[Index]], [[Values]], [[Capability]], [[RemainingElements]], and [[AlreadyCalled]] internal slots.</p>
+    <p>When a `Promise.any` reject element function is called with argument _x_, the following steps are taken:</p>
+    <emu-alg>
+      1. Let _F_ be the active function object.
+      1. Let _alreadyCalled_ be _F_.[[AlreadyCalled]].
+      1. If _alreadyCalled_.[[Value]] is *true*, return *undefined*.
+      1. Set _alreadyCalled_.[[Value]] to *true*.
+      1. Let _index_ be _F_.[[Index]].
+      1. Let _values_ be _F_.[[Values]].
+      1. Let _promiseCapability_ be _F_.[[Capability]].
+      1. Let _remainingElementsCount_ be _F_.[[RemainingElements]].
+      1. Set _values_[_index_] to _x_.
+      1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] - 1.
+      1. If _remainingElementsCount_.[[Value]] is 0, then
+        1. Let _valuesArray_ be CreateArrayFromList(_values_).
```

Comment:
Probably should be something like
```
        Let _valuesArray_ be AggregateError(_values_, ???).
```
I'm not sure

Reply:
We don't yet have a convention for how to specify an error object which has a required property. I think I would do something like

```
1. Let _valuesArray_ be CreateArrayFromList(_values_).
1. Let _exception_ be a newly created AggregateError object.
1. Perform ! CreateDataProperty(_exception_, `"errors"`, _errorsArray_).
1. Return ? Call(*undefined*, _promiseCapability_.[[Reject]], &laquo; _exception_ &raquo;).
```

The "a newly created WhateverError object" phrasing is [used elsewhere](https://tc39.github.io/ecma262/#sec-throw-an-exception). Doing it this way hopefully makes it clear that engines are required to provide `.errors` but are free to make `message` whatever they feel is appropriate, just as with other errors thrown by builtins.